devito/ir/clusters/algorithms.py

```diff
@@ -238,7 +238,7 @@ def guard(clusters):
                 if cd._factor is not None:
                     k = d
                 else:
-                    dims = pull_dims(cd.condition)
+                    dims = pull_dims(cd.condition, flag=False)
```

Comment:
I think this should be `flag=cd.indirect` for the case where the conditional is used as the indexing dimension